### DISCUSSION TITLE
PKPD model for 2drugs

### DISCUSSION THREAD

**Discussion by BoburaII** [LAPKB/Pmetrics#169] [Q&A] [answered]
Hi, 
I am working on a joint PKPD model for simultaneous estimation of PK and PD parameters for two orally administered drugs (A & B). The input and output equations (2 PK equations and 1 joint PD equation) are indicated below:

model file
#pri
Ka, 0, 10
V, 0, 100
KCP, 0, 10
KPC, 0, 10
Vp, 0, 250
CL, 0, 15
Ka1, 0, 5
V1, 0, 900
KCP1, 0, 10
KPC1, 0, 10
Vp1, 0, 800
CL1, 0, 30
Ks, 0, 5
Qmax, 250, 600
IC, 300, 550
H, 0, 20
EC50, 0, 100
H1, 0, 10
EC50_1, 0, 3

#ini
X(7) = IC

#cov
age
sex
weight

#dif
XP(1) = -Ka*X(1)
XP(2) = Ka*X(1) - ((CL/V)+KCP)*X(2) + KPC*X(3)
XP(3) = KCP*X(2) - KPC*X(3)
XP(4) = -Ka1*X(4)
XP(5) = Ka1*X(4) - ((CL1/V1)+KCP1)*X(5) + KPC1*X(6)
XP(6) = KCP1*X(5) - KPC1*X(6)
XP(7) = (Ks*(1-(X(7)/Qmax))* X(7) * (((X(2)/V)**H/(EC50**H + (X(2)/V)**H)) + ((X(5)/V1)**H1/(EC50_1**H1 + (X(5)/V1)**H1))))

#out
Y(1)=X(2)/V
Y(2)=X(5)/V1
Y(3)=X(7)

#err
G=5
0.01, 0.1, 0, 0
0.1, 0.05, 0, 0
0.05, 0.01, 0, 0

Differential block: PK equations 1-3 for drugA; 4-6 for drugB and a joint PD equation 7. On execution of the model, it only picks up the input for drugA. I’ve attempted a work around to allow for input from drugB with the following modification: 

#diff. 
XP(1) = R(1) - Ka*X(1)
XP(4) = R(2) - Ka1*X(4)

Inputs from both drugs now recognised. However, with R(1) and R(2), I’m likely introducing another issue of bolus inputs for drugs which were orally administered.
Any suggestions on how to appropriately implement input from both drugs?

Thanks.

**Comment by mnneely** [comment, marked as answer]
We are shortly going to change this to be more transparent.   Bolus (oral) inputs are handled outside the differential equations so it is currently a different block. The new way will simply have the bolus inputs included in the equations. But for now, include the following block in your model code:

#BOL
NBCOMP(2) = 4

This tells Pmetrics that the bolus for input 2 should go into compartment 4, based on your differential equations. By default, input 1 goes into compartment 1, so you don't need to specify this.

**Reply by BoburaII** [reply]
Many thanks for the insightful response, adding a #BOL block sorts it out. Exciting to hear of developments in the pipeline for future implementation of bolus input as part of diff. equations.
